backend/src/document.ts

```diff
@@ -83,4 +85,51 @@ function getDocumentMetas(folder: string) {
 	return documentMetas;
 }
 
-export { getCommonDocuments, getLevelDocuments, getSandboxDocumentMetas };
+// store vectorised documents for each level as array
+const vectorisedDocuments = (() => {
+	let docs: DocumentsVector[] = [];
+	return {
+		get: () => docs,
+		set: (newDocs: DocumentsVector[]) => {
+			docs = newDocs;
+		},
+	};
+})();
+
+// create and store the document vectors for each level
+async function initDocumentVectors() {
+	const docVectors: DocumentsVector[] = [];
+	const commonDocuments = await getCommonDocuments();
+
+	const levelValues = Object.values(LEVEL_NAMES)
+		.filter((value) => !isNaN(Number(value)))
+		.map((value) => Number(value));
+
+	for (const level of levelValues) {
+		const allDocuments = commonDocuments.concat(await getLevelDocuments(level));
+
+		// embed and store the splits - will use env variable for API key
+		const embeddings = new OpenAIEmbeddings();
+		const docVector = await MemoryVectorStore.fromDocuments(
+			allDocuments,
+			embeddings
+		);
+		// store the document vectors for the level
+		docVectors.push({
+			level,
+			docVector,
+		});
+	}
+	vectorisedDocuments.set(docVectors);
```

Comment:
This is the only place we are calling the setter, so, no need to export the entire vectorisedDocuments object, we only need the getter - safety first. Something like this?

```
const vectorisedDocuments = (() => {
	let docs: DocumentsVector[] = [];
	return {
		get: (level: LEVEL_NAMES) => docs[level]?.docVector,
		set: (newDocs: DocumentsVector[]) => {
			docs = newDocs;
		},
	};
})();
const getDocumentVectors = vectorisedDocuments.get;

...

export {
  getDocumentVectors,
}
```

Note also you won't need to export the existing getLevelDocuments or getCommonDocuments functions as they're only used in this file now, in initDocumentVectors. Might therefore be worth inlining the calls and removing the functions:

```
const commonDocuments = await getDocuments(getFilepath('common'));

for (const level of levelValues) {
  const allDocuments = commonDocuments.concat(await getDocuments(getFilepath(level)));
```

Reply:
@chriswilty 

Because we removed getLevelDocuments and getCommonDocuments, I've deleted the unit tests for those. But now we're left with `initDocumentVectors` being pretty untestable, because it's function is to set the documentVectors variable. So I guess that we can test that it's set correctly. But its type is from langchain and will probably be pretty horrible to test. Any suggestions? Leave it be?

Reply:
Seems pretty low priority